packages/hawtio/src/plugins/camel/CamelContent.tsx

```diff
@@ -47,11 +46,90 @@ type NavItem = {
+// For more info check: https://github.com/hawtio/hawtio-next/issues/237
+const allNavItems: NavItem[] = [
+  { id: 'attributes', title: 'Attributes', component: <Attributes />, isApplicable: isDefaultApplicable },
+  { id: 'operations', title: 'Operations', component: <Operations />, isApplicable: isDefaultApplicable },
+  { id: 'contexts', title: 'Contexts', component: <Contexts />, isApplicable: camelService.isContextsFolder },
+  { id: 'routes', title: 'Routes', component: <CamelRoutes />, isApplicable: camelService.isRoutesFolder },
+  { id: 'endpoints', title: 'Endpoints', component: <Endpoints />, isApplicable: camelService.isEndpointsFolder },
+  {
+    id: 'routeDiagram',
+    title: 'Route Diagram',
+    component: <RouteDiagram />,
+    isApplicable: camelService.canViewRouteDiagram,
+  },
+  {
+    id: 'source',
+    title: 'Source',
+    component: <Source />,
+    isApplicable: camelService.canViewSource,
+  },
+  { id: 'properties', title: 'Properties', component: <Properties />, isApplicable: camelService.hasProperties },
+  {
+    id: 'send',
+    title: 'Send',
+    component: <SendMessage />,
+    isApplicable: camelService.canSendMessage,
+  },
+  {
+    id: 'browse',
+    title: 'Browse',
+    component: <BrowseMessages />,
+    isApplicable: camelService.canBrowseMessages,
+  },
+  {
+    id: 'endpoint-stats',
+    title: 'Endpoints (in/out)',
+    component: <EndpointStats />,
+    isApplicable: camelService.canViewEndpointStats,
+  },
+  { id: 'exchanges', title: 'Exchanges', component: <Exchanges />, isApplicable: camelService.hasExchange },
+  {
+    id: 'rest-services',
+    title: 'Rest Services',
+    component: <RestServices />,
+    isApplicable: camelService.hasRestServices,
+  },
+  {
+    id: 'type-converters',
+    title: 'Type Converters',
+    component: <TypeConverters />,
+    isApplicable: camelService.hasTypeConverter,
+  },
+  { id: 'chart', title: 'Chart', component: <Chart />, isApplicable: isDefaultApplicable },
+  // Applicable for same criteria as trace
+  { id: 'profile', title: 'Profile', component: <Profile />, isApplicable: camelService.canTrace },
+  { id: 'trace', title: 'Trace', component: <Trace />, isApplicable: camelService.canTrace },
+  { id: 'debug', title: 'Debug', component: <Debug />, isApplicable: camelService.canGetBreakpoints },
+]
 export const CamelContent: React.FunctionComponent = () => {
   const { selectedNode } = useContext(CamelContext)
   const routeDiagramContext = useRouteDiagramContext()
-  const { pathname, search } = useLocation()
+  const [activeTab, setActiveTab] = useState<string>('')
+  const [tabItems, setTabItems] = useState<NavItem[]>([])
+
+  useEffect(() => {
+    if (selectedNode) {
+      /* Filter the nav items to those applicable to the selected node */
+      const tabItems = allNavItems.filter(nav => nav.isApplicable(selectedNode))
+      setActiveTab(tabItems[0]?.id ?? '')
```

Comment:
it appears always the first tab is selected. shouldn't we use pathname?

Reply:
Yes you're right, but this was the always the case actually with the camel plugin. I will fix that though.